### PR TITLE
add dz96

### PR DESCRIPTION
**Pull request by kb-elmo**
## Description

Added the DZ96 keyboard. VIA keymap included.

This board is matrix identical to the Espectro however I think it should be added as it's own board since it is extremely confusing for people that have the Tofu96 by KBDfans to not be able to find their board on QMK and it created issues with almost bricked boards in the past already (people tend to just use whatever keyboard they think fits if they can't find the correct one). 
And since the Espectro is not manufactured anymore anyways and is different in the way that it has RGB underglow it makes sense to have the DZ96 separate.

Also ported the board to VIA.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz96/config.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x445A
+#define PRODUCT_ID      0xDB96
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    DZTECH
+#define PRODUCT         DZ96
+#define DESCRIPTION     TOFU96 PCB
```
Comment (fauxpark) [inline keyboards/dztech/dz96/config.h:27]:
```suggestion
```

File: keyboards/dztech/dz96/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ),
+
+    [1] = LAYOUT_default(
+        RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MUTE, KC_VOLD,    KC_VOLU, KC_MNXT, KC_HOME, KC_END,  \
+        _______, RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, _______, _______, _______, _______,          KC_DEL,     _______, _______, _______, _______, \
+        _______, BL_TOGG, BL_INC,  BL_DEC,  _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,    _______, _______, _______,          \
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______, \
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______           \
```
Comment (fauxpark) [inline keyboards/dztech/dz96/keymaps/default/keymap.c:37]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,    KC_SLCK, KC_PAUS, KC_PGUP, KC_PGDN,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,           KC_BSPC,    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_BSLS,    KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                   KC_ENT,     KC_P4,   KC_P5,   KC_P6,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,      KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT,    KC_P0,   KC_PDOT
    ),

    [1] = LAYOUT_default(
        RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MUTE, KC_VOLD,    KC_VOLU, KC_MNXT, KC_HOME, KC_END,
        _______, RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, _______, _______, _______, _______,          KC_DEL,     _______, _______, _______, _______,
        _______, BL_TOGG, BL_INC,  BL_DEC,  _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,    _______, _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______
```

File: keyboards/dztech/dz96/keymaps/iso/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ),
+
+    [0] = LAYOUT_iso(
+        RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MUTE, KC_VOLD,    KC_VOLU, KC_MNXT, KC_HOME,  KC_END, \
+        _______, RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, _______, _______, _______, _______,          KC_DEL,     _______, _______, _______, _______, \
+        _______, BL_TOGG, BL_INC,  BL_DEC,  _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______,          \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______, \
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______           \
```
Comment (fauxpark) [inline keyboards/dztech/dz96/keymaps/iso/keymap.c:36]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,    KC_HOME, KC_END,  KC_PGUP, KC_PGDN,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,           KC_BSPC,    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,                      KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS,          KC_ENT,     KC_P4,   KC_P5,   KC_P6,
        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,      KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT,    KC_P0,   KC_PDOT
    ),

    [0] = LAYOUT_iso(
        RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MUTE, KC_VOLD,    KC_VOLU, KC_MNXT, KC_HOME,  KC_END,
        _______, RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, _______, _______, _______, _______,          KC_DEL,     _______, _______, _______, _______,
        _______, BL_TOGG, BL_INC,  BL_DEC,  _______, _______, _______, _______, _______, _______, _______, _______, _______,                      _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______
```

File: keyboards/dztech/dz96/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    ),
+
+    [3] = LAYOUT_default(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,    _______, _______, _______,          \
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______, \
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______           \
```
Comment (fauxpark) [inline keyboards/dztech/dz96/keymaps/via/keymap.c:55]:
```suggestion
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,    KC_SLCK, KC_PAUS, KC_PGUP, KC_PGDN,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,           KC_BSPC,    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_BSLS,    KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,                   KC_ENT,     KC_P4,   KC_P5,   KC_P6,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,      KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT,    KC_P0,   KC_PDOT
    ),

    [1] = LAYOUT_default(
        RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MUTE, KC_VOLD,    KC_VOLU, KC_MNXT, KC_HOME, KC_END,
        _______, RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, _______, _______, _______, _______,          KC_DEL,     _______, _______, _______, _______,
        _______, BL_TOGG, BL_INC,  BL_DEC,  _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,    _______, _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______
    ),

    [2] = LAYOUT_default(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,    _______, _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______
    ),

    [3] = LAYOUT_default(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,    _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,    _______, _______, _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,    _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______,    _______, _______
```

File: keyboards/dztech/dz96/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# DZ96
+
+![DZ96](https://kbdfans.com/products/tofu96-mechanical-keyboard-diy-kit)
+
+A 96% PCB made by KBDFans. Sold together with the Tofu96 DIY kit.
+
+Keyboard Maintainer: [kb-elmo](https://github.com/kb-elmo)  
+Hardware Supported: DZ96  
+Hardware Availability: [KBDFans](https://kbdfans.com)
```
Comment (fauxpark) [inline keyboards/dztech/dz96/readme.md:9]:
```suggestion
* Keyboard Maintainer: [kb-elmo](https://github.com/kb-elmo)
* Hardware Supported: DZ96
* Hardware Availability: [KBDFans](https://kbdfans.com)
```

File: keyboards/dztech/dz96/rules.mk [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/dztech/dz96/rules.mk:11]:
```suggestion
```

File: keyboards/dztech/dz96/rules.mk [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/dztech/dz96/rules.mk:28]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/dztech/dz96/readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# DZ96
+
+![DZ96](https://kbdfans.com/products/tofu96-mechanical-keyboard-diy-kit)
```
Comment (fauxpark) [inline keyboards/dztech/dz96/readme.md:3]: This needs to be an actual image (preferably hosted on imgur).

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dztech/dz96/keymaps/iso/keymap.c [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    [0] = LAYOUT_iso(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,    KC_HOME, KC_END,  KC_PGUP, KC_PGDN,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,           KC_BSPC,    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,                      KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS,          KC_ENT,     KC_P4,   KC_P5,   KC_P6,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,      KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT,    KC_P0,   KC_PDOT
+    ),
+
+    [0] = LAYOUT_iso(
```
Comment (drashna) [inline keyboards/dztech/dz96/keymaps/iso/keymap.c:30]:
```suggestion
    [1] = LAYOUT_iso(
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dztech/dz96/readme.md (hunk 8 above)
Comment (kb-elmo) [reply keyboards/dztech/dz96/readme.md]: added one.
the only image that I could find is the tofu96 kit photo.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]
I'll probably want to pull out the `espectro` from the codebase at a later point, and move its keymaps over here, and I also don't like the layout macro names (which are copied from `espectro`), so I'll probably change that too (but make them backwards compatible).

Projects for future-me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**